pkg/ottl/ottlfuncs/func_replace_all_patterns.go

```diff
@@ -57,9 +57,47 @@ func ReplaceAllPatterns[K any](target ottl.PMapGetter[K], mode string, regexPatt
 			case modeKey:
 				if compiledPattern.MatchString(key) {
 					updatedKey := compiledPattern.ReplaceAllLiteralString(key, replacement)
-					updated.PutStr(updatedKey, originalValue.Str())
+					switch originalValue.Type() {
```

Comment:
I can't look at this in depth till Wednesday but there should be a way to use pcommon.Value and .CopyTo/generic Put to avoid a switch statement.

Reply:
@Ishmeet I believe this will work:

```go
if compiledPattern.MatchString(key) {
	updatedKey := compiledPattern.ReplaceAllLiteralString(key, replacement)
	originalValue.CopyTo(updated.PutEmpty(updatedKey))
} else {
	originalValue.CopyTo(updated.PutEmpty(key))
}
```

Reply:
yeah